src/main/java/sg/ncl/MainController.java

```diff
@@ -5016,4 +5020,184 @@ private TeamQuota extractTeamQuotaInfo(String responseBody) {
+        jsonObject.put("emails", jsonArray.toString());
+
+        HttpEntity<String> request = createHttpEntityWithBody(jsonObject.toString());
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity responseEntity = null;
+        try {
+            responseEntity = restTemplate.exchange(properties.addMemberByEmail(teamId), HttpMethod.POST, request, String.class);
+        } catch (RestClientException e) {
+            log.warn("Error connecting to sio team service for adding members by email: {}", e);
+            redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+            return "redirect:add_member";
+        }
+
+        String responseBody = responseEntity.getBody().toString();
+
+        if (RestUtil.isError(responseEntity.getStatusCode())) {
+            MyErrorResource error;
+            try {
+                error = objectMapper.readValue(responseBody, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+                switch (exceptionState) {
+                    case INVALID_EMAIL_ADDRESS_EXCEPTION:
+                        log.warn("Adding members by emails: Invalid email address");
+                        redirectAttributes.addFlashAttribute(MESSAGE, "Email address is not valid");
+                        break;
+                    default:
+                        log.warn("Adding members by emails : sio or deterlab adapter connection error");
+                        // possible sio or adapter connection fail
+                        redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+                        break;
+                }
+                return "redirect:add_member";
+            } catch (IOException e) {
+                throw new WebServiceRuntimeException(e.getMessage());
+            }
+        }
+        log.info("Adding members by email for team {} successful", teamId);
+        redirectAttributes.addFlashAttribute(MESSAGE_SUCCESS, "New members have been added successful!");
+        return "redirect:add_member";
+    }
+
+
+    // when user clicks password reset link in the email
+    @GetMapping(path = "/newMemberResetPassword", params = {"uid", "key"})
+    public String resetPasswordForNewMember(@NotNull @RequestParam("uid") final String uid,
+                                    @NotNull @RequestParam("key") final String key,
+                                              Model model) {
+        NewClassMemberPasswordResetForm newClassMemberPasswordResetForm = new NewClassMemberPasswordResetForm();
+        newClassMemberPasswordResetForm.setKey(key);
+        newClassMemberPasswordResetForm.setUid(uid);
+
+        model.addAttribute("newClassMemberPasswordResetForm", newClassMemberPasswordResetForm );
+        // redirect to the page for user to enter new password
+        return "new_member_reset_password";
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "new_member_reset_password" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)